File: net/day1/Dial.cs
public enum DialDirection
{
    Left = 0,
    Right = 1
}

public class Dial
{
    int _dialPosition;
    int _dialMaximimumPosition;

    public int DialPosition
    {
        get => _dialPosition;
    }

    public Dial(int startinPosition = 0, int dialMaximimumPosition = 99)
    {
        if (startinPosition < 0 || startinPosition >= dialMaximimumPosition)
        {
            throw new ArgumentException($"{nameof(startinPosition)} has to be >= 0 && < dialMaximumPosition");
        }

        _dialPosition = startinPosition;
        _dialMaximimumPosition = dialMaximimumPosition;
    }

    public void MoveDial(DialDirection direction)
    {
        int newDialPosition = _dialPosition;

        if (direction == DialDirection.Left) newDialPosition -= 1;
        else newDialPosition += 1;

        if (newDialPosition > _dialMaximimumPosition) newDialPosition = 0;
        else if (newDialPosition < 0) newDialPosition = _dialMaximimumPosition;

        _dialPosition = newDialPosition;
    }
}

File: rust/day1/src/dial.rs
pub enum DialDirection
{
    Left,
    Right
}

pub struct Dial
{
    pub dial_position: i32,
    pub dial_max_position: i32
}

impl Dial
{
    pub fn move_dial(&mut self, direction: &DialDirection)
    {
        let mut new_dial_position = self.dial_position;

        if matches!(direction, DialDirection::Left) { new_dial_position -= 1; }
        else { new_dial_position += 1; }
        
        if new_dial_position > self.dial_max_position { new_dial_position = 0; }
        else if new_dial_position < 0 { new_dial_position = self.dial_max_position; }

        self.dial_position = new_dial_position;
    }
}